File: javascript/word-count/word-count.js
export const countWords = (sentence) => {
	//const regex = /\w+(?:[']?\w+)*/g;
	const regex = /\w+([']?\w+)*/g;
	const words = sentence.toLowerCase().match(regex);
	let wordMap = {};

	for (let i = 0; i < words.length; i++) {	
		wordMap[words[i]] = typeof wordMap[words[i]] === 'undefined' ? 1 : wordMap[words[i]] + 1;
	}

	return wordMap;
};

File: python/word-count/word_count.py
import re
from collections import Counter


def count_words(sentence: str) -> dict[str, int]:
    words = re.findall("[a-z0-9]+(?:'[a-z]+)?", sentence.lower())
    word_counts = Counter(words)

    return {item: count for item, count in word_counts.most_common()}